internal/httpcli/client.go

```diff
@@ -41,6 +41,14 @@ type Doer interface {
 	Do(*http.Request) (*http.Response, error)
 }
 
+type MockDoer struct {
+	DoFunc func(*http.Request) (*http.Response, error)
+}
+
+func (m MockDoer) Do(req *http.Request) (*http.Response, error) {
+	return m.DoFunc(req)
+}
+
```

Comment:
minor drive by, but you get slicely nicer callsites using a type alias to a function. For example this is what `http.HandlerFunc` does
```suggestion
type MockDoer func(*http.Request) (*http.Response, error)

func (m MockDoer) Do(req *http.Request) (*http.Response, error) {
	return m(req)
}
```

Reply:
Thanks for this, Keegan! Didn't know that, appreciate the tip!